common/navbar.js

```diff
@@ -284,6 +284,59 @@
 
                     scope.onLinkClick = onLinkClick(ConfigUtils, logService, UriUtils, $window);
 
+                    scope.showRidSearch = function () {
+                        return chaiseConfig.resolverImplicitCatalog !== null && chaiseConfig.hideSearchRID !== true
+                    }
+
+                    // RID search turned off in the cases of:
+                    //  - resolverImplicitCatalog == null
+                    //  - OR hideSearchRID == true
+                    //
+                    // The following cases need to be handled for the resolverImplicitCatalog value:
+                    //  - if catalogId is defined:
+                    //    - if resolverImplicitCatalog === null:        turn off config
+                    //    - if resolverImplicitCatalog === catalogId:   /id/RID
+                    //    - otherwise:                                  /id/catalogId/RID
+                    //  - else:
+                    //    - if resolverImplicitCatalog === null:        turn off config
+                    //    - if resolverImplicitCatalog is Not a Number: /id/RID
+                    //    - otherwise:                                  /id/resolverImplicitCatalog/RID
+                    scope.ridSearch = function () {
+                        var resolverId = chaiseConfig.resolverImplicitCatalog,
+                            url = "/id/", catId, splitId;
+
+                        if (isCatalogDefined(catalogId)) {
+                            splitId = UriUtils.splitVersionFromCatalog(catalogId);
+
+                            // use `/id/catalog/ridSearchTerm` format if:
+                            //   - resolver id is a different catalog id than current page
+                            //   - resolver id is NaN and !null
+                            if (isNaN(resolverId) || resolverId !== catalogId) {
```

Comment:
In `resolvePermalink` this check is done by `==`. So to be consistent let's change this to `!=`. Then we will support both string and integer in both places